src/parse-templates.ts

```diff
@@ -17,11 +17,23 @@ export interface TemplateLiteralMatch {
   end: RegExpMatchArray;
 }
 
+/**
+ * Represents a static import of a template literal.
+ *
+ * @param importPath the path to the package from which we want to import the template literal (e.g.: 'ember-cli-htmlbars')
+ * @param importIdentifier the name of the template literal (e.g.: 'hbs') or 'default' if this package exports a default function
+ */
 export interface StaticImportConfig {
   importPath: string;
   importIdentifier: string;
 }
 
+/**
+ * The input options to instruct parseTemplates on how to parse the input.
+ *
+ * @param templateTag tag if parsing template tags is enabled
+ * @param templateLiteral an array of StaticImportConfig to indicate which static imports are expected in this template
+ */
```

Comment:
Similarly:

```ts
/**
 * The input options to instruct parseTemplates on how to parse the input.
 *
 * @param templateTag 
 * @param templateLiteral 
 */
export interface ParseTemplatesOptions {
  /** Tag to use, if parsing template tags is enabled. */
  templateTag?: string;
  /** Which static imports are expected in this template. */
  templateLiteral?: StaticImportConfig[];
}
```

Reply:
Non-blocking: whenever I see two optional things like this, it makes me think it should *probably* actually be a union type where *neither* is optional in their own type:

```ts
type TemplateTagOption = {
  /** Tag to use, if parsing template tags is enabled. */
  templateTag: string;
}

type TemplateLiteralOption = {
  /** Which static imports are expected in this template. */
  templateLiteral: StaticImportConfig[];
}

type ParseTemplatesOptions = TemplateTagOption | TemplateLiteralOption;
```

The idea is: it's completely useless to pass `{}` here: it doesn't *do* anything and doesn't even inform what you mean on the caller side. So, instead, using a(n optional) union type says the actual intent, which is "if you pass something it has to be one or the other of these"—at least, if I’ve understood it correctly and it's not legitimate to pass *both*.

Reply:
@chriskrycho my expectations is that we can actually send _both_, as I'm not sure if we can easily determine whether to expect the file will use template tags or template literal. One possible improvement would be to make `templateLiteral` _required_, as it can just be an empty array. Thoughts?

Reply:
I’m fine with it if they actually are both optional! It’s just not usually the case. Not blocking though, to be sure.